### ISSUE TITLE
NotImplementedError: The validate_template action has not been implemented

### ISSUE THREAD

**Issue by alekna** [spulec/moto#876]
moto `0.4.31`

```======================================================================
ERROR: test_create_on_update (tests.test_cf.TestStackActions)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/lauris/git/<...>/env/lib/python3.5/site-packages/moto/core/models.py", line 71, in wrapper
    result = func(*args, **kwargs)
  File "/home/lauris/git/<...>/tests/test_cf.py", line 69, in test_create_on_update
    self.config, update=True, create_on_update=True)
  File "/home/lauris/git/<...>/<...>/cf.py", line 197, in create_stack
    valid_template, err = validate_template(conn, tpl_body, tpl_url)
  File "/home/lauris/git/<...>/<...>/cf.py", line 248, in validate_template
    conn.validate_template(template_body=template_body, template_url=template_url)
  File "/home/lauris/git/<...>/env/lib/python3.5/site-packages/boto/cloudformation/connection.py", line 808, in validate_template
    verb="POST")
  File "/home/lauris/git/<...>/env/lib/python3.5/site-packages/boto/connection.py", line 1192, in get_object
    response = self.make_request(action, params, path, verb)
  File "/home/lauris/git/<...>/env/lib/python3.5/site-packages/boto/connection.py", line 1116, in make_request
    return self._mexe(http_request)
  File "/home/lauris/git/<...>/env/lib/python3.5/site-packages/boto/connection.py", line 944, in _mexe
    response = connection.getresponse()
  File "/usr/lib64/python3.5/http/client.py", line 1194, in getresponse
    response = self.response_class(self.sock, method=self._method)
  File "/home/lauris/git/<...>/env/lib/python3.5/site-packages/boto/connection.py", line 390, in __init__
    http_client.HTTPResponse.__init__(self, *args, **kwargs)
  File "/usr/lib64/python3.5/http/client.py", line 235, in __init__
    self.fp = sock.makefile("rb")
  File "/home/lauris/git/<...>/env/lib/python3.5/site-packages/httpretty/core.py", line 332, in makefile
    self._entry.fill_filekind(self.fd)
  File "/home/lauris/git/<...>/env/lib/python3.5/site-packages/httpretty/core.py", line 591, in fill_filekind
    status, headers, self.body = self.callable_body(self.request, self.info.full_url(), headers)
  File "/home/lauris/git/<...>/env/lib/python3.5/site-packages/moto/core/responses.py", line 107, in dispatch
    return cls()._dispatch(*args, **kwargs)
  File "/home/lauris/git/<...>/env/lib/python3.5/site-packages/moto/core/responses.py", line 167, in _dispatch
    return self.call_action()
  File "/home/lauris/git/<...>/env/lib/python3.5/site-packages/moto/core/responses.py", line 193, in call_action
    raise NotImplementedError("The {0} action has not been implemented".format(action))
NotImplementedError: The validate_template action has not been implemented```

**Comment by spulec** [comment]
Thanks for opening this. It will be easy enough to add a function that does a basic pass, but I'm imagining it will be tough to implement the real thing.

Will keep this open as a feature request. Thanks

**Comment by alekna** [comment]
Please let me know when you add a basic pass.

**Comment by JackDanger** [comment]
@alekna would you be interested in trying to implement this? Even if you only get a little way it's a great way to learn more about the (simple!) internals of Moto. I'd be happy to join you and help writing the tests and completing this feature.

**Comment by alekna** [comment]
Will do,

**Comment by dbfr3qs** [comment]
Was any progress made on this?

**Comment by alekna** [comment]
AFAIK, no.

**Comment by adamdavis40208** [comment]
@spulec super old thread. What are your thoughts for mocking this?

An easy pass would be "is it valid json/yaml, does it at least meet some of the basic requirements of cloudformation (must include one resource, etc)"

Setting out with a goal of "perfect offline cloudformation validation tool", while i'm sure the community would want it, would be pretty insane to implement + to keep up to date with how often cloudformation changes.

**Comment by spulec** [comment]
Works for me

**Comment by bblommers** [comment]
Closing - this was implemented in #1921
